futures-util/src/io/mod.rs

```diff
@@ -324,3 +322,61 @@ pub trait AsyncWriteExt: AsyncWrite {
 }
 
 impl<W: AsyncWrite + ?Sized> AsyncWriteExt for W {}
+
+/// An extension trait which adds utility methods to `AsyncBufRead` types.
+pub trait AsyncBufReadExt: AsyncBufRead {
```

Comment:
I imagine we'll also want `read_line`, `lines` etc. but those can wait for a followup